### PR TITLE
Adding an API to enable the drain mode

### LINKED ISSUE #6268: Consume the drain mode api exposed by the webjobs sdk
Related to https://github.com/Azure/azure-webjobs-sdk/issues/2527

### PR DESCRIPTION
**Pull request by soninaren**
Documentation present in the issue below.
https://github.com/Azure/azure-webjobs-sdk/issues/2527

resolves #6268 

### Pull request checklist

* [ ] My changes **do not** require documentation changes
    * [x] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
    * [x] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by pr: https://github.com/Azure/azure-functions-host/pull/6338
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information
Additionally performed manual testing to understand behavior of extensions and have filed relevant issues on the individual extensions repo. Also tested thing is co-ordination with the Glenna and divya on private stamps.
related issues linked here: https://github.com/Azure/azure-functions-host/issues/5672#issuecomment-649052092

### CODE REVIEW

File: src/WebJobs.Script.WebHost/Controllers/HostController.cs [hunk 1]
```diff
@@ -101,6 +101,19 @@ public async Task<IActionResult> GetHostStatus([FromServices] IScriptHostManager
             return Ok(status);
         }
 
+        [HttpGet]
+        [HttpPost]
+        [Route("admin/host/drain")]
+        [Authorize(Policy = PolicyNames.AdminAuthLevelOrInternal)]
+        public IActionResult Drain([FromServices] IDrainModeManager drainModeManager)
+        {
+            _logger.LogInformation("Received request for draining host");
```
Comment (pragnagopa) [inline src/WebJobs.Script.WebHost/Controllers/HostController.cs]: log debug?

**Comment by pragnagopa** [comment]
@soninaren - PR does not include tests. Can you please add unit tests or update description to include what tests you ran to vertfy the fix?

**Comment by soninaren** [comment]
> @soninaren - PR does not include tests. Can you please add unit tests or update description to include what tests you ran to vertfy the fix?

All the unit tests for this API are in webjobs sdk. There is nothing is happening inside function host for us except for calling the api exposed from job host.
Here is the webjobs sdk PR with unit tests, https://github.com/Azure/azure-webjobs-sdk/pull/2526/files

**Comment by soninaren** [comment]
Pending: Need to update the webjobs sdk, package update

File: src/WebJobs.Script.WebHost/Controllers/HostController.cs (hunk 1 above)
Comment (soninaren) [reply src/WebJobs.Script.WebHost/Controllers/HostController.cs]: done

**Review by divyagandhisethi — DISMISSED** [review]

File: src/WebJobs.Script.WebHost/Controllers/InstanceController.cs [hunk 2]
```diff
@@ -1,10 +1,12 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
+using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs.Host;
```
Comment (divyagandhisethi) [inline src/WebJobs.Script.WebHost/Controllers/InstanceController.cs]: you can remove this as well.

**Review by divyagandhisethi — APPROVED** [review]

**Review by pragnagopa — APPROVED** [review]
Nit: I briefly looked at the WebJobs related changes. Can you please open a follow up item to change logs to debug?

**Comment by soninaren** [comment]
> Nit: I briefly looked at the WebJobs related changes. Can you please open a follow up item to change logs to debug?
I have fixed this in the current PR.

Are you referring to update the log level to debug for web jobs changes?

**Pull request merged**